src/test/kotlin/org/rust/cargo/toolchain/CargoTest.kt

```diff
@@ -8,10 +8,22 @@ package org.rust.cargo.toolchain
 import com.intellij.execution.configurations.GeneralCommandLine
 import com.intellij.openapi.util.SystemInfo
 import com.intellij.util.net.HttpConfigurable
+import org.junit.Test
 import org.rust.lang.RsTestBase
+import org.rust.openapiext.pathAsPath
 import java.nio.file.Paths
 
 class CargoTest : RsTestBase() {
+
+    fun `test run arguments preserved`() = checkCommandLine(
+        cargo
+            .toColoredCommandLine(CargoCommandLine("run", listOf("--bin", "parity", "--", "--prune", "archive")))
+            .withCharset(Charsets.UTF_8),
+        "cmd: /usr/bin/cargo run --color=always --bin parity -- --prune archive\n" +
+            "env: RUSTC=/usr/bin/rustc, RUST_BACKTRACE=full, TERM=ansi",
+        ""
+    )
```

Comment:
Excellent, thanks!

I am afraid this'll fail on windows though (that's what the second argument means). See the tests below to find out how this second string should look like.

Oh, and while we are at it, perhaps it would be prudent to write a couple of unit-tests for `getRunArguments` and `getBuildArguments` as well? Nothing fancy, just constructing a `CargoCommandLine` with `listOf("--bin", "parity", "--", "--prune", "archive")` and checking result exactly.